drivers/volume/linstor/linstor.go

```diff
@@ -0,0 +1,412 @@
+	// try getting the provisioner from the Storage class.
+	if val, ok := pvc.Annotations[pvcProvisionerAnnotation]; ok {
+		provisioner = val
+	} else {
+		storageClassName := k8shelper.GetPersistentVolumeClaimClass(pvc)
+		if storageClassName != "" {
+			storageClass, err := storage.Instance().GetStorageClass(storageClassName)
+			if err == nil {
+				provisioner = storageClass.Provisioner
+			} else {
+				logrus.Warnf("Error getting storageclass %v for pvc %v: %v", storageClassName, pvc.Name, err)
+			}
+		}
+	}
+
+	if provisioner == "" {
+		// Try to get info from the PV since storage class could be deleted
+		pv, err := core.Instance().GetPersistentVolume(pvc.Spec.VolumeName)
+		if err != nil {
+			logrus.Warnf("Error getting pv %v for pvc %v: %v", pvc.Spec.VolumeName, pvc.Name, err)
+			return false
+		}
+		logrus.Infof("No information in PVC, try PV?")
+		return l.OwnsPV(pv)
+	}
+
+	if provisioner == provisionerName {
+		logrus.Infof("-> yes")
+		return true
+	}
+
+	logrus.Infof("-> no")
+	return false
+}
+
+func (l *linstor) OwnsPV(pv *v1.PersistentVolume) bool {
+	logrus.Infof("called: OwnsPV(%s)", pv.ObjectMeta.Name)
+	provisioner := ""
+	// Check the annotation in the PV for the provisioner
+	if val, ok := pv.Annotations[pvProvisionedByAnnotation]; ok {
+		provisioner = val
+	}
+	if provisioner == provisionerName {
+		logrus.Infof("-> yes (PV)")
+		return true
+	}
+	logrus.Infof("-> no (PV)")
+	return false
+}
+
+func (l *linstor) GetSnapshotPlugin() snapshotVolume.Plugin {
+	logrus.Infof("called: GetSnapshotPlugin()")
+	return nil
+}
+
+func (l *linstor) GetSnapshotType(snap *snapv1.VolumeSnapshot) (string, error) {
+	logrus.Infof("called: GetSnapshotType(%s)", snap.String())
+	return "", &errors.ErrNotImplemented{}
+}
+
+func (l *linstor) Stop() error {
+	logrus.Infof("called: Stop()")
+	close(l.stopChannel)
+	return nil
+}
+
+func randString(n int) string {
+	const letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+func (l *linstor) GetClusterID() (string, error) {
+	logrus.Infof("called: GetClusterID()")
+	props, err := l.cli.Nodes.GetControllerProps(context.TODO())
+	if err != nil {
+		return "", fmt.Errorf("failed to query linstor controller properties: %w", err)
+	}
+
+	var id string
+	for k, v := range props {
+		if k == "Aux/StorkClusterId" {
+			id = v
+			break
+		}
+	}
+
+	if id == "" {
+		id = randString(16)
+		logrus.Infof("linstor: no cluster ID found, generating new (%s)", id)
+		err := l.cli.Nodes.ModifyController(context.TODO(), lclient.GenericPropsModify{
+			OverrideProps: lclient.OverrideProps{
+				"Aux/StorkClusterId": id,
+			},
+		})
+		if err != nil {
+			return "", fmt.Errorf("failed to set cluster ID property: %w", err)
+		}
+	}
+	return id, nil
+}
+
+func init() {
+	l := &linstor{}
+	err := l.Init(nil)
```

Comment:
Instead of calling `Init` from here call it from each API that requires to use the linstor client if the driver has not been already initialized. 
Otherwise, we'll always start the nodecache in this driver even though it might not be used. The portworx driver also doesn't call `Init` from here anymore